google/resource_pubsub_subscription.go

```diff
@@ -21,14 +25,16 @@ func resourcePubsubSubscription() *schema.Resource {
 			},
 
 			"topic": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
 
 			"ack_deadline_seconds": &schema.Schema{
```

Comment:
There's an opportunity to specify the default here instead of in the create codepath below (I see below it's 10; does that match the GCP api?).

Reply:
I prefer to leave it as `Computed` right now to avoid creating diffs for our users (if for instance they updated this field from the UI). A diff right now forces the creation of a new resources. We don't support update. I will fix this once we add update support in #455. I added a comment about this in #455

Reply:
Ah, yeah, that sounds like a much better experience. 👍